pkg/framework/context/context.go

```diff
@@ -83,20 +83,20 @@ func (p *Provider) InboundTransportEndpoint() string {
 
 // InboundMessageHandler return inbound message handler
 func (p *Provider) InboundMessageHandler() transport.InboundMessageHandler {
-	return func(payload []byte) error {
+	return func(envelope *wallet.Envelope) error {
```

Comment:
Somehow I'm hesitant about the need to send the entire envelope, since these really ought to be two different layers.

Side note: it seems weird that the Envelope type is defined by Wallet @Baha-sk @Moopli

Reply:
If service handle needs to access envelope then InboundMessageHandler need entire envelope.

Alternative solution is to pass second argument next to payload to pass 'ToVerKeys',  but arguments list may grow in future as per requirement. --> func(payload, toVerKeys)

or we can a sub-struct inside 'wallet.Envelope' something like 'Keys' and using that struct here along with payload --> func(payload, keys)

Reply:
> Is DID exchange the only protocol that will care about VerKeys? @rolsonquadras @llorllale ?
@troyronda the primary reason for this PR is https://github.com/hyperledger/aries-framework-go/issues/398#issue-501672749 - to fetch the invitation based on recKey. For now it looks to be specific to DIDExchange.

Reply:
created #474 for envelope type issue

Reply:
> Somehow I'm hesitant about the need to send the entire envelope, since these really ought to be two different layers.
> 
> Is DID exchange the only protocol that will care about VerKeys? @rolsonquadras @llorllale ?
> 
> Side note: it seems weird that the Envelope type is defined by Wallet @Baha-sk @Moopli

this envelope a wallet/packager envelope type that has the payload and verification keys. Which is different than the one representing a JWE found here: https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/crypto/jwe/authcrypt/authcrypt.go#L55

in my PR, it has moved to a new `envelope` package: https://github.com/hyperledger/aries-framework-go/pull/459/files#diff-2e40038f150e7f436acca5a5949cb54fR46

Reply:
We should followup with a better abstraction (rather than passing the entire envelope to a layer that shouldn't need it).